### PR TITLE
Generate init script

### PR DESCRIPTION
**Pull request by ofalvai**
### Checklist

- [ ] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- [ ] `step.yml` and `README.md` is updated with the changes (if needed)

### Version

Requires NO [version update](https://semver.org/)

### Changes

- Generate Gradle init script based on a template by inserting the version number, API endpoint, auth token
- Add the init script to `~/.gradle/init.gradle` so that it's applied to any subsequent Gradle build
- Add various tests

### Decisions

The API endpoint is not final yet, this will come in a follow-up PR.

### CODE REVIEW

File: step/step_test.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+				inputParser:    stepconf.NewInputParser(tt.envRepo),
+				commandFactory: command.NewFactory(tt.envRepo),
+				envRepo:        tt.envRepo,
+				pathModifier:   fakePathModifier{path: fakePath},
+			}
+			err := step.Run()
+			if (err != nil) != tt.wantErr {
+				t.Errorf("Run() error = %v, wantErr %v", err, tt.wantErr)
+			} else if err == nil {
+				_, err := ioutil.ReadFile(path.Join(fakePath, "init.gradle"))
```
Comment (hisaac) [inline step/step_test.go:50]: GoLand is telling me that the `ReadFile` method is deprecated. It links to [this documentation page](https://pkg.go.dev/io/ioutil#ReadFile), which says:

> Deprecated: As of Go 1.16, this function simply calls os.ReadFile.

Think we should switch it?

```suggestion
				_, err := os.ReadFile(path.Join(fakePath, "init.gradle"))
```

**Review by hisaac — APPROVED** [review]
I added one question with an optional suggestion, but I'm fine either way if you want to leave it as-is. Looking good!

File: step/step_test.go (hunk 1 above)
Comment (ofalvai) [reply step/step_test.go]: Huh, I was wondering why I didn't get this suggestion from GoLand. Then I figured it out: I'm using Go 1.17.11 (the lowest version still installed on our stacks), where it's just a note without the `Deprecated:` prefix :)

Comment (ofalvai) [reply step/step_test.go]: Anyway, I replaced the function call, take another look!

Comment (hisaac) [reply step/step_test.go]: Ah yep, I'm running Go 19.0.1 currently.

**Review by hisaac — APPROVED** [review]

**Pull request merged**